cmd/helpers/readChannels.go

```diff
@@ -28,13 +29,23 @@ func tailLineToChan(lines chan *tail.Line, batchSize int) <-chan []extractor.BSt
 				}
 				batch = append(batch, extractor.BString(line.Text))
 				if len(batch) >= batchSize {
-					output <- batch
+					output <- extractor.InputBatch{
```

Comment:
Does consolidating this to a closure work for performance?

Reply:
Did some benchmarking.. wrapping it in a function so deep in the call can cost in terms of ops (~ 1ns/op, which is a 400% increase). Since this code isn't too duplicative, leaving as is for now